src/domain/tradingview/useTVDatafeed.ts

```diff
@@ -86,108 +95,195 @@ export default function useTVDatafeed({ dataProvider }: Props) {
+          onHistoryCallback(bars, { noData });
+        } catch {
+          onErrorCallback("Unable to load historical data!");
+        }
       },
-    };
-  }, [chainId, stableTokens, supportedResolutions]);
+      async subscribeBars(
+        symbolInfo: SymbolInfo,
+        resolution: ResolutionString,
+        onRealtimeCallback: SubscribeBarsCallback
+      ) {
+        await subscribeBars({
+          symbolInfo,
+          resolution,
+          onRealtimeCallback,
+          chainId,
+          missingBarsInfoRef,
+          feedDataRef,
+          lastBarTimeRef,
+          intervalRef,
+          supportedResolutions,
+          tvDataProviderRef,
+        });
+      },
+      unsubscribeBars: (id) => {
+        // id is in the format ETH_#_USD_#_5
+        const ticker = id.split("_")[0];
+        const isStable = stableTokens.includes(ticker);
+        if (!isStable && intervalRef.current) {
+          clearInterval(intervalRef.current);
+        }
+      },
+    },
+  };
+}
+
+export function subscribeBars({
+  symbolInfo,
+  resolution,
+  onRealtimeCallback,
+  chainId,
+  feedDataRef,
+  intervalRef,
+  supportedResolutions,
+  tvDataProviderRef,
+  lastBarTimeRef,
+  missingBarsInfoRef,
+}: {
+  symbolInfo: Pick<SymbolInfo, "ticker" | "isStable">;
+  resolution: ResolutionString;
+  onRealtimeCallback: SubscribeBarsCallback;
+  chainId: number;
+  supportedResolutions: { [key: string]: string };
+  tvDataProviderRef: MutableRefObject<TVDataProvider | undefined>;
+  intervalRef: MutableRefObject<number | undefined>;
+  missingBarsInfoRef: MutableRefObject<{
+    bars: FromOldToNewArray<Bar>;
+    isFetching: boolean;
+  }>;
+  feedDataRef: MutableRefObject<boolean>;
+  lastBarTimeRef: MutableRefObject<number>;
+}) {
+  const period = supportedResolutions[resolution];
+  const { ticker, isStable } = symbolInfo;
+  if (!ticker) {
+    return;
+  }
+
+  if (intervalRef.current !== undefined) {
+    clearInterval(intervalRef.current);
+    intervalRef.current = undefined;
+  }
+
+  const handleInterval = () => {
+    if (missingBarsInfoRef.current.isFetching || !feedDataRef.current) return;
+    if (missingBarsInfoRef.current.bars?.length > 0) {
+      let processedBarTimes: FromOldToNewArray<number> = [];
+      missingBarsInfoRef.current.bars.forEach((bar: Bar, index) => {
+        if (processedBarTimes.includes(bar.time)) return;
+        if (index !== 0 && bar.time < processedBarTimes.at(-1)!) {
+          console.error(
+            "missingBarsInfoRef bars order is violating TradingView api schema. See https://www.tradingview.com/charting-library-docs/latest/connecting_data/Datafeed-API#subscribebars"
+          );
+        }
+
+        onRealtimeCallback(formatTimeInBarToMs(bar));
+        processedBarTimes.push(bar.time);
+      });
+      missingBarsInfoRef.current.bars = [];
+    } else {
+      tvDataProviderRef.current?.getLiveBar(chainId, ticker, period).then((bar) => {
```

Comment:
we call this line by 500ms interval. Can't it be that `getLiveBar` responds in more than 500ms and thus would mess up bars?

Reply:
but it later checks that 
```
          bar.ticker === tvDataProviderRef.current?.currentTicker &&
          bar.period === tvDataProviderRef.current?.currentPeriod
```
it seems to take that into acc?

Reply:
I will add
```
bar.time >= lastBarTimeRef.current
```
but im not sure it will do anything

Reply:
try intentionally slowdown getLiveBar to ~1 second resp time. If it doesn't affect badly than it's fine like it's